### ISSUE TITLE
Link for source of is_x86_feature_detected is broken

### ISSUE THREAD

**Issue by hellow554** [rust-lang/rust#96506]
IIRC there is a broken link checker, but somehow it doesn't catch this case:

https://doc.rust-lang.org/std/macro.is_x86_feature_detected.html

![grafik](https://user-images.githubusercontent.com/921462/165701381-b75e3674-54fd-4903-83c1-3cc903061d3f.png)

https://doc.rust-lang.org/1.60.0/src/std_detect/detect/arch/x86.rs.html#18-197 which is a 404

@rustbot modify labels: A-docs

**Comment by hellow554** [comment]
Also, all source links of the macros in https://doc.rust-lang.org/nightly/std/arch/index.html are broken as well.

**Comment by jyn514** [comment]
I think the right fix is to pass -Zrustdoc-map to cargo when documenting the standard library. We already do that for the compiler docs: https://github.com/rust-lang/rust/blob/96834f0231277e8feb8dcf185b2af082ad2e39f6/src/bootstrap/doc.rs#L698

mentoring instructions: add that same flag to `impl Step for doc::Std`.

**Comment by jyn514** [comment]
cc https://github.com/rust-lang/rust/issues/81311, which is one way to fix this in rustdoc, but will result in a worse experience for the source view.

**Comment by ismailarilik** [comment]
I just opened a PR for this issue: https://github.com/rust-lang/rust/pull/131067
I confirmed that it fixes this issue. One of my concerns is that it is very similar to [the closed PR](https://github.com/rust-lang/rust/pull/96519) even if I followed the mentoring instructions.

And the other concern is that although this PR fixes the main issue, there is another issue: Macros here seems still not working: https://doc.rust-lang.org/nightly/std/arch/index.html#macros Actually `is_x86_feature_detected` is working after this PR but others aren't; I think there is some mismatch about generating links between Cargo and docs.rs.

I can investigate this second issue further in this issue or in a new issue but at first I wanna ensure that I am on the right way.
